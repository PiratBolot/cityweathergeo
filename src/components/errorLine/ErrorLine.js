import React from 'react';
import {connect} from "react-redux";
import './ErrorLine.css'

class ErrorLine extends React.Component {
    render() {
        return (
            <div>
                {this.props.errorMessage.isError &&
                <div className="error_msg">{this.props.errorMessage.errorMsg}</div>
                }
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    errorMessage: {isError: state.errorMessage.isError, errorMsg: state.errorMessage.errorMsg}
});

export default connect(mapStateToProps)(ErrorLine);
