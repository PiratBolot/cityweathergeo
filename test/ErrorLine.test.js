import React from 'react';
import {mount} from '../enzyme/enzyme';
import {applyMiddleware, createStore} from "redux";
import {Provider} from "react-redux";
import rootReducer from "../src/reducers/Reducers";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";
import ErrorLine from "../src/components/errorLine/ErrorLine";

describe('Error Line', () => {
    const initialState = {
        trackedCities: ['Syktyvkar', 'Moscow', 'Kiev'],
        errorMessage: {isError: true, errorMsg: "Tests crushed"},
        geolocation: {}
    };
    let store,wrapper;
    beforeEach(()=>{
        store = createStore(rootReducer, initialState, applyMiddleware(thunk));
        wrapper = mount( <Provider store={store}><ErrorLine /></Provider> );
    })
    it('+++capturing Snapshot of ErrorLine', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});