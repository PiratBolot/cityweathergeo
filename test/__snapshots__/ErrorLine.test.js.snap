// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Line +++capturing Snapshot of ErrorLine 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ErrorLine)>
    <ErrorLine
      dispatch={[Function]}
      errorMessage={
        Object {
          "errorMsg": "Tests crushed",
          "isError": true,
        }
      }
    >
      <div>
        <div
          className="error_msg"
        >
          Tests crushed
        </div>
      </div>
    </ErrorLine>
  </Connect(ErrorLine)>
</Provider>
`;
